core/src/main/scala/me/wojnowski/oidc4s/IdTokenVerifier.scala

```diff
@@ -125,17 +118,24 @@ object IdTokenVerifier {
       private def ensureNotExpired(now: Instant, expiresAt: Instant): Either[Error.TokenExpired, Unit] =
         Either.raiseWhen(expiresAt.isBefore(now))(TokenExpired(since = expiresAt))
 
-      private def decodeHeader(headerJson: String): Either[CouldNotDecodeHeader, JoseHeader] =
+      private def decodeHeader(headerJson: String): Either[Error, JoseHeader] = {
+        val unsupportedAlgorithmPrefix = "Unsupported algorithm: "
```

Comment:
Perhaps a nitpick, but I'd keep the responsibility of this string in a single place. What do you think about having something like `JoseHeaderCirceDecoder.checkUnsupportedAlgorithm(errorMessage: String)` and just use it here?

Reply:
Circe is in a separate module, so it would have to be in `core`.

In the end, I encapsulated the prefix and "logic" within `UnsupportedAlgorithm`'s `toRawError` and `fromRawError` functions. This way it looks less scary in the core logic, I think 👻

Reply:
That's even better!

LGTM🚀